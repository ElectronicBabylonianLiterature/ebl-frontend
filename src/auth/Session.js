import applicationScopes from './applicationScopes.json'
import { Set } from 'immutable'

class Session {
  constructor (accessToken, idToken, expiresAt, scopes) {
    this.accessToken = accessToken
    this.idToken = idToken
    this.expiresAt = expiresAt
    this.scopes = Set(scopes)
    Object.freeze(this)
  }

  isAuthenticated () {
    return new Date().getTime() < this.expiresAt
  }

  getAccessToken () {
    if (this.accessToken) {
      return this.accessToken
    } else {
      throw new Error('No access token')
    }
  }

  hasScope (scope) {
    return this.isAuthenticated() && this.scopes.has(scope)
  }

  isAllowedToReadWords () {
    return this.hasApplicationScope('readWords')
  }

  isAllowedToWriteWords () {
    return this.hasApplicationScope('writeWords')
  }

  isAllowedToReadFragments () {
    return this.hasApplicationScope('readFragments')
  }

  isAllowedToTransliterateFragments () {
    return this.hasApplicationScope('transliterateFragments')
  }

  isAllowedToLemmatizeFragments () {
    return this.hasApplicationScope('lemmatizeFragments')
  }

  hasBetaAccess () {
    return this.hasApplicationScope('accessBeta')
  }

  hasApplicationScope (applicationScope) {
    const scope = applicationScopes[applicationScope]
    return this.hasScope(scope)
  }
}

export default Session
