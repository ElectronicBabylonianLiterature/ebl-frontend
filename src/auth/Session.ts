import Folio from 'fragmentarium/domain/Folio'
import applicationScopes from './applicationScopes.json'

export interface Session {
  isAllowedToReadWords(): boolean

  isAllowedToWriteWords(): boolean

  isAllowedToReadFragments(): boolean

  isAllowedToTransliterateFragments(): boolean

  isAllowedToLemmatizeFragments(): boolean

  isAllowedToAnnotateFragments(): boolean

  isAllowedToReadBibliography(): boolean

  isAllowedToWriteBibliography(): boolean

  isAllowedToReadTexts(): boolean

  isAllowedToWriteTexts(): boolean

  hasBetaAccess(): boolean

  isAllowedToReadFolio(folio: Folio): boolean

  isGuestSession(): boolean
}

class GuestSession implements Session {
  isAllowedToReadWords(): boolean {
    return true
  }

  isAllowedToWriteWords(): boolean {
    return false
  }

  isAllowedToReadFragments(): boolean {
    return true
  }

  isAllowedToTransliterateFragments(): boolean {
    return false
  }

  isAllowedToLemmatizeFragments(): boolean {
    return false
  }

  isAllowedToAnnotateFragments(): boolean {
    return false
  }

  isAllowedToReadBibliography(): boolean {
    return true
  }

  isAllowedToWriteBibliography(): boolean {
    return false
  }

  isAllowedToReadTexts(): boolean {
    return true
  }

  isAllowedToWriteTexts(): boolean {
    return false
  }

  hasBetaAccess(): boolean {
    return false
  }

  isAllowedToReadFolio(folio: Folio): boolean {
    return folio.isOpen
  }

  isGuestSession(): boolean {
    return true
  }
}

export const guestSession = new GuestSession()

export default class MemorySession implements Session {
  private readonly scopes: ReadonlySet<string>

  constructor(scopes: readonly string[]) {
    this.scopes = new Set(scopes)
  }

  isAllowedToReadWords(): boolean {
    return this.hasApplicationScope('readWords')
  }

  isAllowedToWriteWords(): boolean {
    return this.hasApplicationScope('writeWords')
  }

  isAllowedToReadFragments(): boolean {
    return this.hasApplicationScope('readFragments')
  }

  isAllowedToTransliterateFragments(): boolean {
    return this.hasApplicationScope('transliterateFragments')
  }

  isAllowedToLemmatizeFragments(): boolean {
    return this.hasApplicationScope('lemmatizeFragments')
  }

  isAllowedToAnnotateFragments(): boolean {
    return this.hasApplicationScope('annotateFragments')
  }

  isAllowedToReadBibliography(): boolean {
    return this.hasApplicationScope('readBibliography')
  }

  isAllowedToWriteBibliography(): boolean {
    return this.hasApplicationScope('writeBibliography')
  }

  isAllowedToReadTexts(): boolean {
    return this.hasApplicationScope('readTexts')
  }

  isAllowedToWriteTexts(): boolean {
    return this.hasApplicationScope('writeTexts')
  }

  hasBetaAccess(): boolean {
    return this.hasApplicationScope('accessBeta')
  }

  isAllowedToReadFolio(folio: Folio): boolean {
    return folio.isOpen || this.hasApplicationScope(`read${folio.name}Folios`)
  }

  isGuestSession(): boolean {
    return false
  }

  hasApplicationScope(applicationScope: string): boolean {
    const scope = applicationScopes[applicationScope]
    return this.scopes.has(scope)
  }
}
