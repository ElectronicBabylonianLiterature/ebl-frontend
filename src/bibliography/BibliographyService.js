import Promise from 'bluebird'

export default class BibliographyService {
  #bibliographyRepository

  constructor(bibliographyRepository) {
    this.#bibliographyRepository = bibliographyRepository
  }

  create(entry) {
    return this.#bibliographyRepository.create(entry)
  }

  find(id) {
    return this.#bibliographyRepository.find(id)
  }

  update(entry) {
    return this.#bibliographyRepository.update(entry)
  }

  search(query) {
    const queryRegex = /^([^\d]+)(?: (\d{4})(?: (.*))?)?$/
    const match = queryRegex.exec(query)
    return match
      ? this.#bibliographyRepository.search(
          match[1],
          match[2] || '',
          match[3] || ''
        )
      : Promise.resolve([])
  }
}
