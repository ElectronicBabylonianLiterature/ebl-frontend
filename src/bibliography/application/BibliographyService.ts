import Promise from 'bluebird'
import BibliographyEntry from 'bibliography/domain/BibliographyEntry'

export interface BibliographySearch {
  search(query: string): Promise<readonly BibliographyEntry[]>
}

export default class BibliographyService implements BibliographySearch {
  private readonly bibliographyRepository

  constructor(bibliographyRepository) {
    this.bibliographyRepository = bibliographyRepository
  }

  create(entry: BibliographyEntry): Promise<BibliographyEntry> {
    return this.bibliographyRepository.create(entry)
  }

  find(id: string): Promise<BibliographyEntry> {
    return this.bibliographyRepository.find(id)
  }

  update(entry: BibliographyEntry): Promise<BibliographyEntry> {
    return this.bibliographyRepository.update(entry)
  }

  search(query: string): Promise<readonly BibliographyEntry[]> {
    const queryRegex = /^([^\d]+)(?: (\d{4})(?: (.*))?)?$/
    const match = queryRegex.exec(query)
    return match
      ? this.bibliographyRepository.search(
          match[1],
          match[2] || '',
          match[3] || ''
        )
      : Promise.resolve([])
  }
}
