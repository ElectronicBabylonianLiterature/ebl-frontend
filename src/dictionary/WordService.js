class WordService {
  constructor(wordRepository) {
    this.wordRepository = wordRepository
  }

  find(id) {
    return this.wordRepository.find(id)
  }

  search(query) {
    return this.wordRepository.search(query)
  }

  update(word) {
    return this.wordRepository.update(word)
  }
}

export default WordService
