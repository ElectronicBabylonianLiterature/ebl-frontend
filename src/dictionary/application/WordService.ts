import Promise from 'bluebird'
import Word from 'dictionary/domain/Word'
import WordRepository from 'dictionary/infrastructure/WordRepository'
import _ from 'lodash'
import { stringify } from 'query-string'

export interface WordQuery {
  word?: string
  meaning?: string
  root?: string
  vowelClass?: string
}

class WordService {
  private readonly wordRepository: WordRepository

  constructor(wordRepository: WordRepository) {
    this.wordRepository = wordRepository
  }

  find(id: string): Promise<Word> {
    return this.wordRepository.find(id)
  }

  findAll(ids: string[]): Promise<readonly Word[]> {
    return this.wordRepository.findAll(ids)
  }

  search(query: WordQuery): Promise<Word[]> {
    return this.wordRepository.search(
      stringify(query, { skipEmptyString: true })
    )
  }

  searchLemma(lemma: string): Promise<readonly Word[]> {
    return _.isEmpty(lemma)
      ? Promise.resolve([])
      : this.wordRepository.searchLemma(lemma)
  }

  update(word: Word): Promise<Word> {
    return this.wordRepository.update(word)
  }
}

export default WordService
