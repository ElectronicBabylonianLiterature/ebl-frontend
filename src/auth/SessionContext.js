import React from 'react'
import Session from './Session'

const SessionContext = React.createContext(new Session('', '', 0, []))

export default SessionContext
