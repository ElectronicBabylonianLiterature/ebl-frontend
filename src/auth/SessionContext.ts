import React, { Context } from 'react'
import { guestSession, Session } from 'auth/Session'

const SessionContext: Context<Session> = React.createContext<Session>(
  guestSession
)

export default SessionContext
