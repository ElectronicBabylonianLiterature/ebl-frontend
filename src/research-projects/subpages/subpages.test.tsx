import React from 'react'
import Promise from 'bluebird'
import WordService from 'dictionary/application/WordService'
import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
import FragmentService from 'fragmentarium/application/FragmentService'
import CaicPage from './caic'
import BibliographyService from 'bibliography/application/BibliographyService'
import { render } from '@testing-library/react'
import { createMemoryHistory, MemoryHistory } from 'history'
import { Router, withRouter } from 'react-router-dom'
import Session from 'auth/Session'
import SessionContext from 'auth/SessionContext'
import MemorySession from 'auth/Session'

jest.mock('fragmentarium/application/FragmentService')
jest.mock('auth/Session')
jest.mock('fragmentarium/application/FragmentSearchService')
jest.mock('bibliography/application/BibliographyService')
jest.mock('dictionary/application/WordService')

let fragmentService: jest.Mocked<FragmentService>
let fragmentSearchService: jest.Mocked<FragmentSearchService>
let wordService: jest.Mocked<WordService>
let bibliographyService: jest.Mocked<BibliographyService>
let container: HTMLElement
let session: jest.Mocked<Session>
let history: MemoryHistory

async function renderProjectPage() {
  const CaicPageWithRouter = withRouter<any, typeof CaicPage>(CaicPage)
  container = render(
    <Router history={history}>
      <SessionContext.Provider value={session}>
        <CaicPageWithRouter
          fragmentService={fragmentService}
          fragmentSearchService={fragmentSearchService}
          wordService={wordService}
          bibliographyService={bibliographyService}
        />
      </SessionContext.Provider>
    </Router>
  ).container
}

beforeEach(async () => {
  fragmentService = new (FragmentService as jest.Mock<
    jest.Mocked<FragmentService>
  >)()
  fragmentSearchService = new (FragmentSearchService as jest.Mock<
    jest.Mocked<FragmentSearchService>
  >)()
  wordService = new (WordService as jest.Mock<jest.Mocked<WordService>>)()
  bibliographyService = new (BibliographyService as jest.Mock<
    jest.Mocked<BibliographyService>
  >)()
  history = createMemoryHistory()
  session = new (MemorySession as jest.Mock<jest.Mocked<MemorySession>>)()
  session.isAllowedToReadFragments.mockReturnValue(true)
  fragmentService.fetchPeriods.mockReturnValue(Promise.resolve([]))
  fragmentService.fetchGenres.mockReturnValue(Promise.resolve([]))
  wordService.findAll.mockReturnValue(Promise.resolve([]))
})

describe('Project pages', () => {
  it('displays CAIC page', async () => {
    await renderProjectPage()
    expect(container).toMatchSnapshot()
  })
})
