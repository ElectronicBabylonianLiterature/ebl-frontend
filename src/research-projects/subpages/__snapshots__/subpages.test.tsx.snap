// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project pages displays CAIC page 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="main__content"
    >
      <header
        class="main__header"
      >
        <nav
          aria-label="breadcrumb"
          class="main__breadcrumbs"
        >
          <ol
            class="breadcrumb"
          >
            <li
              class="breadcrumb-item active"
            >
              <a
                href="/"
              >
                eBL
              </a>
            </li>
            <li
              aria-current="page"
              class="breadcrumb-item active"
            >
              Projects
            </li>
            <li
              aria-current="page"
              class="breadcrumb-item active"
            >
              CAIC
            </li>
          </ol>
        </nav>
        <h2
          class="main__heading"
        >
          Cuneiform Artefacts of Iraq in Context
        </h2>
        <div
          class="main__toolbar btn-toolbar"
          role="toolbar"
        />
      </header>
      <div
        class="container-fluid"
      >
        <h3>
          Introduction
        </h3>
        <p>
          The cuneiform artifacts of the Iraq Museum in Baghdad are a central part of the cultural heritage of Mesopotamia, which is of great importance to all of humanity. As an interdisciplinary academic project, the aim of CAIC is to document, edit, and analyze approximately 17,000 cuneiform tablets both from historical and linguistic perspectives. With the help of the latest digital approaches it will also make them available online to the general public and experts from various disciplines.
        </p>
        <h3>
          Search CAIC Texts
        </h3>
        <form
          class=""
        >
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="number"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col"
            >
              <input
                aria-label="Number"
                class="form-control"
                id="number"
                name="number"
                placeholder="Museum, accession, CDLI, or excavation number"
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              >
                At least one of prefix, number or suffix must be specified.
              </div>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="reference"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col"
            >
              <div
                class=" css-2b097c-container"
              >
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-g1d714-ValueContainer"
                  >
                    <div
                      class=" css-1wa3eu0-placeholder"
                    >
                      Name Year Title
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="Select bibliography reference"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-sm-5"
            >
              <input
                aria-label="Pages"
                class="form-control"
                id="reference"
                name="pages"
                placeholder="Page"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="period"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col"
            >
              <div
                class="script-selection__selection css-2b097c-container"
              >
                <div
                  class="period-selector__control css-yk16xz-control"
                >
                  <div
                    class="period-selector__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="period-selector__placeholder css-1wa3eu0-placeholder"
                    >
                      Period modifier
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="period-selector__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="select-period-modifier"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-3-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="period-selector__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="period-selector__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="period-selector__indicator period-selector__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="script-selection__selection css-2b097c-container"
              >
                <div
                  class="period-selector__control css-yk16xz-control"
                >
                  <div
                    class="period-selector__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="period-selector__placeholder css-1wa3eu0-placeholder"
                    >
                      Period
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="period-selector__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="select-period"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-5-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="period-selector__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="period-selector__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="period-selector__indicator period-selector__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="genre"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col"
            >
              <div
                class=" css-2b097c-container"
              >
                <div
                  class="genre-selector__control css-yk16xz-control"
                >
                  <div
                    class="genre-selector__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="genre-selector__placeholder css-1wa3eu0-placeholder"
                    >
                      Genre
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="genre-selector__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="select-genre"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-6-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="genre-selector__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="genre-selector__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="genre-selector__indicator genre-selector__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="lemmas"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col"
            >
              <div
                class=" css-2b097c-container"
              >
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-g1d714-ValueContainer"
                  >
                    <div
                      class=" css-1wa3eu0-placeholder"
                    >
                      Lemmata
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="Select lemmata"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-7-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-sm-3"
            >
              <div
                class="script-selection__selection css-2b097c-container"
              >
                <div
                  class=" css-yk16xz-control"
                >
                  <div
                    class=" css-g1d714-ValueContainer"
                  >
                    <div
                      class=" css-1uccc91-singleValue"
                    >
                      Same line
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class=""
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="Select lemma query type"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-4-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 form-label"
              for="transliteration"
            >
              <i
                class="fas fa-info-circle"
              />
            </label>
            <div
              class="col-sm-10"
            >
              <textarea
                aria-label="Transliteration"
                class="form-control"
                id="transliteration"
                name="transliteration"
                placeholder="Transliterations"
                rows="2"
              />
            </div>
          </div>
        </form>
        <div
          class="btn-toolbar"
          role="toolbar"
        >
          <div
            class="SearchForm__ButtonToolbar col-sm offset-sm-2"
          >
            <button
              class="w-25 m-1 btn btn-primary"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
